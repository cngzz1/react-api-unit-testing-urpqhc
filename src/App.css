.App {
	text-align: center;
	background-color: #282c34;
	height: auto;
	color: white;
	font-size: 20px;
}
