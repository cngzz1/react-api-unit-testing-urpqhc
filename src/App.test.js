import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";
import "./App.css";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { mount, shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Header from "./components/Header";
import AppClass from "./App-Class";
configure({ adapter: new Adapter() });

describe("App Component Testing", () => {
	// #0. Snapshot Testing.
	test("Snapshot testing for APP component", () => {
		const AppComponent = renderer
			.create(
				<Router>
					<App />
				</Router>,
			)
			.toJSON();
		expect(AppComponent).toMatchSnapshot();
	});
	// #1.1.Render Whole Application
	test("Render App Component", () => {
		const AppComponent = mount(
			<Router>
				<App />
			</Router>,
		);
	});
	// #1.2. Render Isolated component
	test("Render App Component", () => {
		shallow(<App />);
	});
	// #2.1. Testing the elements
	test("Render h1 element", () => {
		let app = shallow(<App />);
		let hello = <h1>Hello</h1>;
		expect(app.contains(hello)).toEqual(true);
	});
	// #2.2.Test elements
	test("Render p element", () => {
		let app = shallow(<App />);
		expect(app.find("p").text()).toBe("Venkatesh");
	});
	// #2.3.Test sub components
	test("Render Header Component", () => {
		shallow(<Header />);
	});
	// #3.1 Testing state in class components
	test("testing state in class Component", () => {
		let app = shallow(<AppClass />);
		expect(app.state("username")).toBe("Venkatesh");
	});
	// #3.2 Testing state in functional components
	test("testing state in class Component", () => {
		let app = shallow(<App />);
		expect(app.find("p").text()).toBe("Venkatesh");
	});
	// #4.1 Testing state after updating in class components
	test("testing state in class Component", () => {
		let app = shallow(<AppClass />);
		let button = app.find("button");
		button.simulate("click");
		expect(app.find("p").text()).toBe("Vivek");
	});
	// #4.2 Testing state after updating in functional components
	test("testing state in class Component", () => {
		let app = shallow(<App />);
		let button = app.find("button");
		button.simulate("click");
		expect(app.find("p").text()).toBe("Vivek");
	});
	// #5. Debugging and Testing
	test("Debugging", () => {
		let app = shallow(<App />);
		// console.log(app.debug());
		app.find("h1").forEach((heading, index) => {
			if (index === 1) {
				expect(heading.text()).toBe("404 Not Found!");
			}
		});
		expect(app.find("h1").first().text()).toBe("Hello");
	});
});
