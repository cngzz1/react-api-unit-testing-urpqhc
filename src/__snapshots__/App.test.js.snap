// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Testing Snapshot testing for APP component 1`] = `
<div
  className="App"
>
  <nav
    className="navbar bg-dark justify-content-center sticky-top"
  >
    <li
      className="nav-link"
    >
      <a
        className="nav-link"
        href="/"
      >
        Refresh
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/menu"
        onClick={[Function]}
      >
        Sidebar
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/dashboard"
        onClick={[Function]}
      >
        Dashboard
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/projects"
        onClick={[Function]}
      >
        My Projects
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/topics"
        onClick={[Function]}
      >
        My Topics
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/about-lazy-load"
        onClick={[Function]}
      >
        Lazy Loading
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/contact"
        onClick={[Function]}
      >
        Contact Us
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/logout"
        onClick={[Function]}
      >
        Logout
      </a>
    </li>
  </nav>
  <h1>
    Hello
  </h1>
  <p>
    Venkatesh
  </p>
  <button
    onClick={[Function]}
  >
    Update
  </button>
  <div>
    <h1
      style={
        Object {
          "margin": 300,
        }
      }
    >
      üíêWelcome to React Routing Crash Courseüíê
    </h1>
  </div>
</div>
`;
